ccm_web/server/src/api/api.service.ts

```diff
@@ -120,6 +128,35 @@ export class APIService {
     return await sectionHandler.enrollUsers(sectionUsers)
   }
 
+  async enrollSectionExternalUsers (user: User, sectionId: number, sectionUsers: SectionExternalUserDto[]): Promise<string> {
+    // Create all requested users, noting failures
+    const adminRequestor = this.canvasService.createRequestorForAdmin('/api/v1/')
+    const adminHandler = new AdminApiHandler(adminRequestor, user.loginId)
+    const canvasConfig = this.configService.get('canvas', { infer: true })
+    const createUserResponses = await adminHandler.createExternalUsers(sectionUsers, canvasConfig.newUserAccountID)
+    const newUsers = createUserResponses.filter(response => !isAPIErrorData(response)) as CanvasUser[]
```

Comment:
So we're just casting here cause TypeScript isn't smart enough to know the `filter` results in only `CanvasUser[]`. Wish there was another way, but this seems pretty safe...

Reply:
How do we differentiate here between a call that failed because of an API problem versus a call that failed because the user already exists? In similar circumstances, I believe I have short-circuited the full API operation if a connection error occurred on one of multiple calls. This probably needs more discussion. I'm not seeing a good way to report real errors here.

Reply:
Resolving this, should be mostly taken care of now.